src/hsm_cl.rs

```diff
@@ -0,0 +1,134 @@
+use class_group::primitives::cl_dl_lcm::{CLDLProofPublicSetup, Witness, HSMCL, PK};
+use curv::arithmetic::traits::Converter;
+use curv::arithmetic::traits::Samplable;
+use curv::elliptic::curves::traits::ECPoint;
+use curv::elliptic::curves::traits::ECScalar;
+use curv::BigInt;
+use curv::FE;
+use curv::GE;
+
+pub use class_group::primitives::cl_dl_lcm::Ciphertext;
+pub type PublicKey = PK;
+
+pub type Proof = CLDLProofPublicSetup;
+
+lazy_static::lazy_static! {
+     pub static ref PUBLIC_SETUP: BigInt = BigInt::from(b"A2LPoC".as_ref());
```

Comment:
Side note: `lazy_static` actually depends on the unmaintained primitive `spin`: https://github.com/rust-lang-nursery/lazy-static.rs/issues/163

I would suggest using `conquer_once` if you need lazy computation. I think it is better because it doesn't need macros. In any case, this instance of global state seems to be unnecessary.

Reply:
Nod. The solution I went with is just passing it around. The public key shouldn't have state in it so I didn't want to put more stuff in there. So it's just another argument for now.